airbyte_cdk/manifest_server/command_processor/processor.py

```diff
@@ -0,0 +1,151 @@
+import logging
+from typing import Any, Dict, Iterable, List, Mapping, Optional, Tuple
+
+from airbyte_protocol_dataclasses.models import (
+    AirbyteCatalog,
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    Status,
+    TraceType,
+)
+from airbyte_protocol_dataclasses.models import Type as AirbyteMessageType
+from fastapi import HTTPException
+
+from airbyte_cdk.connector_builder.models import StreamRead
+from airbyte_cdk.connector_builder.test_reader import TestReader
+from airbyte_cdk.entrypoint import AirbyteEntrypoint
+from airbyte_cdk.models import (
+    AirbyteStateMessage,
+    ConfiguredAirbyteCatalog,
+)
+from airbyte_cdk.sources.declarative.manifest_declarative_source import (
+    ManifestDeclarativeSource,
+)
+
+
+class ManifestCommandProcessor:
+    _source: ManifestDeclarativeSource
+    _logger = logging.getLogger("airbyte.manifest-server")
+
+    def __init__(self, source: ManifestDeclarativeSource) -> None:
+        self._source = source
+
+    def test_read(
+        self,
+        config: Mapping[str, Any],
+        catalog: ConfiguredAirbyteCatalog,
+        state: List[AirbyteStateMessage],
+        record_limit: int,
+        page_limit: int,
+        slice_limit: int,
+    ) -> StreamRead:
+        """
+        Test the read method of the source.
+        """
+
+        test_read_handler = TestReader(
+            max_pages_per_slice=page_limit,
+            max_slices=slice_limit,
+            max_record_limit=record_limit,
+        )
+
+        stream_read = test_read_handler.run_test_read(
+            source=self._source,
+            config=config,
+            configured_catalog=catalog,
+            state=state,
+            stream_name=catalog.streams[0].stream.name,
+            record_limit=record_limit,
+        )
+
+        return stream_read
+
+    def check_connection(
+        self,
+        config: Mapping[str, Any],
+    ) -> Tuple[bool, str]:
+        """
+        Check the connection to the source.
+        """
+
+        spec = self._source.spec(self._logger)
+        messages = AirbyteEntrypoint(source=self._source).check(spec, config)
+        messages_by_type = self._get_messages_by_type(messages)
+        self._raise_on_trace_message(messages_by_type)
+        connection_status = self._get_connection_status(messages_by_type)
```

Comment:
I believe `EntryPointOutput` already implements most (maybe all?) of these methods.

https://airbytehq.github.io/airbyte-python-cdk/airbyte_cdk/test/entrypoint_wrapper.html#EntrypointOutput

<img width="229" height="548" alt="Image" src="https://github.com/user-attachments/assets/5aa7d6a0-2693-444f-afb9-38300ee742bf" />